### PR TITLE
Forge: Add --cups and --no-rate-limit options

### LINKED ISSUE #3823: Add --cups and --no-rate-limit options to forge
### Component

Forge

### Describe the feature you would like

in anvil we have

`--no_rate_limit` and `--cups` options to control budgets for the provider in forking mode.

we should add this to forge as well, ideally via `Config` in rpc_endpoints settings

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Evalir**
## Motivation
Closes #3823. These options were added to anvil, and should be ported to forge.

## Solution

Adds the options to forge, always prioritizing `--no-rate-limit` over `--cups` whenever the former's set.

### CODE REVIEW

File: config/src/lib.rs [hunk 1]
```diff
@@ -305,6 +305,9 @@ pub struct Config {
     /// Disables storage caching entirely. This overrides any settings made in
     /// `rpc_storage_caching`
     pub no_storage_caching: bool,
+    /// Disables rate limiting entirely. This overrides any settings made in
+    /// `compute_units_per_second`
+    pub no_rate_limit: bool,
```
Comment (mattsse) [inline config/src/lib.rs:310]: this needs a more descriptive name, like `no_rpc_rate_limit`.

**Review by mattsse — APPROVED** [review]
awesome, lgtm

would be great if you could add these to the book as well, can reuuse anvil docs if exist

**Comment by mattsse** [comment]
I've added some aliases here https://github.com/mattsse/foundry/tree/Evalir/master

since I'm not allowed to push to your branch, please merge this into yours

**Comment by mattsse** [comment]
ah I think, we need to mark `no_rpc_rate_limit` as serde(default), since we only writing this into the provider `if self.no_rpc_rate_limit`, so the tests are failing because the key is missing

**Comment by Evalir** [comment]
ahh that makes sense. i was wondering since yesterday why tests were failing this way

**Comment by mattsse** [comment]
but I'm actually not sure, currently checking other settings

**Comment by mattsse** [comment]
@Evalir 
ah check `EvmArgs` it's still named `no_rate_limit` so it's serialized that way.
let just rename this `no_rpc_rate_limit` as well, to make it consistent

so for all usage of `no_rate_limit` + `alias = no_rpc_rate_limit` just flip it to `no_rpc_rate_limit` + alias `no_rate_limit`

**Comment by Evalir** [comment]
Aight that makes sense. Done in https://github.com/foundry-rs/foundry/pull/4134/commits/b0578bd3c9b2e9a3eb9431e5aa61a501e7256414 . Should we go ahead and also change Anvil's flag? or should I do that in another PR?

**Comment by mattsse** [comment]
4) . Should we go ahead and also change Anvil's flag? or should I do that in another PR (preferred imo)?

that's fine, now that we have an alias.

thanks so much for this!

**Comment by Evalir** [comment]
aight the alias will do then np! is the ci hung up btw?

**Comment by mattsse** [comment]
> aight the alias will do then np! is the ci hung up btw?

this seems to be an issue with some infura API keys

can reproduce this independently, perhaps just flaky atm, or need to replace some keys...

**Comment by Evalir** [comment]
bruh, it actually seems that the `can_extract_config_values` test is still failing? weird as there doesn't seem to be anything wrong with it at first. will try out things to get the ci to pass

**Comment by mattsse** [comment]
fixed the issue and pushed changes to https://github.com/mattsse/foundry/tree/Evalir/master

please rebase, I'd appreciate it if you'd consider enabling "Allow edits by maintainer"

**Comment by Evalir** [comment]
Aight rebased! And I already had "Allow edits by maintainer" enabled—were you having problems with my branch for this?

**Comment by mattsse** [comment]
thanks.

ah sorry, I misread the git error message...

**Pull request merged**